packages/serverless-collector/src/activate.js

```diff
@@ -6,32 +6,43 @@
 
 const instanaCore = require('@instana/core');
 const { backendConnector, consoleLogger } = require('@instana/serverless');
-
 const identityProvider = require('./identity_provider');
 
 const { tracing, util: coreUtil } = instanaCore;
 const { normalizeConfig } = coreUtil;
+const customMetrics = require('./metrics');
 
 let logger = consoleLogger;
 
 const config = normalizeConfig({});
 config.logger = logger;
 
-function init() {
+async function init() {
+  // NOTE: We accept for `process.env.INSTANA_DEBUG` any string value - does not have to be "true".
   if (process.env.INSTANA_DEBUG || process.env.INSTANA_LOG_LEVEL) {
     logger.setLevel(process.env.INSTANA_DEBUG ? 'debug' : process.env.INSTANA_LOG_LEVEL);
   }
 
-  // NOTE: This package will not collect metrics.
   // NOTE: This package does not support autotracing.
+  // NOTE: This package does not support metrics.
 
   try {
+    // NOTE: We have to call pre-init because we are running an async call to get the service name.
+    //       The goal is to register our instrumentations as early as possible.
+    //       Otherwise we can easily run into errors e.g. from `hasThePackageBeenInitializedTooLate`.
+    instanaCore.preInit();
+
+    const serviceName = await customMetrics.name(config, logger);
+    if (serviceName) {
+      config.serviceName = serviceName;
+    }
+
     identityProvider.init();
     backendConnector.init(identityProvider, logger, false, true, 950);
     instanaCore.init(config, backendConnector, identityProvider);
     tracing.activate();
 
-    logger.debug('@instana/serverless-collector initialized.');
+    logger.info('@instana/serverless-collector initialized.');
```

Comment:
qs: is it possible to include the version of `@instana/serverless-collector`, that will help us in trouble shooting.